### ISSUE TITLE
folder structure

### ISSUE THREAD

**Issue by j4m3s** [PhpGt/Database#1]
How about the following folder structure for users to organise their sql files in their own project:

```
- src
   L ddl
       L mysql
           L <whatever>
       L sqlite
           L <whatever>
       L <etc>
   L dml
       L mysql
           L <whatever>
       L sqlite
           L <whatever>
       L <etc>
```

**Comment by g105b** [comment]
Is it likely that more than one database engine will be used in the same project?

Either way, this could be configured in the config.ini per engine as an optional directory prefix.

**Comment by j4m3s** [comment]
I'm afraid it is possible, yes - for phpgt modules that could be composered
in.  When I was looking at making the csrf module db-aware I realised that
I could need to create separate db scripts for each possible database.  I
guess writing the scripts using only ansi sql would address that - I can't
remember why I couldn't just do that.
On 20 Apr 2016 22:49, "Greg Bowler" <notifications@github.com> wrote:

Is it likely that more than one database engine will be used in the same
project?

Either way, this could be configured in the config.ini per engine as an
optional directory prefix.

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub
<https://github.com/phpgt/database/issues/1#issuecomment-212622677>

**Comment by g105b** [comment]
I think that configurable paths will be able to deal with that use case just fine. Hopefully I'll get some time soon to implement something and we can do some testing.

**Comment by j4m3s** [comment]
If you implement the query classes as object classes you could have a mysql one (with any mysql extensions available - see ticket #4) which looks in a default, but configurable, folder for its ddl and dql statements. 

Thinking through a real example though: if you make the path to sql files configurable, won't that break any composed-in php.gt modules that have put their sql files in a different location?  Or are you thinking each module would have its own configuration overrides, each of which would be parsed on each page request?

**Comment by g105b** [comment]
See #14.
